docs/run-node/4-register-validator.mdx

```diff
@@ -116,9 +122,105 @@ curl --data '{
 }' http://localhost:9650/ext/info | jq -r ".result.nodeID"
 ```
 
-## Staking
+#### Create a Persistent Staking Directory (Recommended for Docker)
+
+To make it easier to manage your keys with Docker, move them to a dedicated directory outside the default location.
+
+```bash
+# Create a dedicated directory
+sudo mkdir -p /opt/flare/staking
+
+# Move your keys
+sudo mv ~/.avalanchego/staking/staker.key /opt/flare/staking/
+sudo mv ~/.avalanchego/staking/staker.crt /opt/flare/staking/
+```
+
+## Run the node
+
+Now, start your node using command-line flags or environment variables that point to your `staker.key` and `staker.crt` files.
+This ensures your node always starts with the correct `Node-ID`.
+
+### From source
+
+Add the `--staking-tls-cert-file` and `--staking-tls-key-file` flags to your [startup command](/run-node/from-source#run-the-node).
+
+```bash
+# Assumes staking keys are in the default location
+./build/avalanchego --network-id=flare \
+  --http-host= \
+  --bootstrap-ips="$(curl -sX POST --data '{"jsonrpc":"2.0", "id":1, "method":"info.getNodeIP"}' -H 'content-type:application/json;' [https://flare-bootstrap.flare.network/ext/info](https://flare-bootstrap.flare.network/ext/info) | jq -r '.result.ip')" \
+  --bootstrap-ids="$(curl -sX POST --data '{"jsonrpc":"2.0", "id":1, "method":"info.getNodeID"}' -H 'content-type:application/json;' [https://flare-bootstrap.flare.network/ext/info](https://flare-bootstrap.flare.network/ext/info) | jq -r '.result.nodeID')" \
+  # highlight-next-line
+  --staking-tls-cert-file="~/.avalanchego/staking/staker.crt" \
```

Comment:
In the context of building from source, since the keys have been moved in the previous step to `/opt/flare/staking/` these should be `/opt/flare/staking/staker.crt` and `/opt/flare/staking/staker.key`